### PR TITLE
Fix Monitor test

### PR DESCRIPTION
**Pull request by PaulCalabrese**
Add new DCPSMonitor config flag to fix Monitor test and make initialization easier.
Enhance Monitor test to really check the functionality is operational.

### CODE REVIEW

File: tests/DCPS/Monitor/run_test.pl [hunk 1]
```diff
@@ -80,6 +84,13 @@
 
 unlink $dcpsrepo_ior;
 
+my $mon_count = `grep -c Report mon.out`;
```
Comment (mitza-oci) [inline tests/DCPS/Monitor/run_test.pl]: This should be able to run on systems without 'grep'.  Equivalent functionality takes just a few lines of perl.

File: dds/monitor/MonitorFactoryImpl.cpp [hunk 2]
```diff
@@ -177,6 +177,7 @@ MonitorFactoryImpl::initialize()
   DDS::DataWriterQos dw_qos;
   publisher->get_default_datawriter_qos(dw_qos);
   dw_qos.durability.kind = DDS::TRANSIENT_LOCAL_DURABILITY_QOS;
+  dw_qos.history.kind = DDS::KEEP_ALL_HISTORY_QOS;
```
Comment (mitza-oci) [inline dds/monitor/MonitorFactoryImpl.cpp]: This may increase resources needed by the monitor library.  Is it needed?

**Comment by PaulCalabrese** [comment]
I'll fix the grep usage. I didn't mean to leave that in this form.

The monitor test was seeing a high variability in the number of monitoring
samples received. Things got much more consistent after I added the
KEEP_ALL on both sides. I am open to suggestions for what would be an
appropriate combination of settings (maybe set max_samples_per_instance?).



Paul

**Comment by mitza-oci** [comment]
Does the test need to count the number of samples?  Would the number of instances be more useful?

**Pull request merged**